modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/PicklistObjectFieldBusinessType.java

```diff
@@ -177,9 +170,23 @@ public void predefineObjectFieldSettings(
 			ObjectStateFlow objectStateFlow =
 				objectFieldSetting.getObjectStateFlow();
 
-			_objectStateFlowLocalService.addObjectStateFlow(
-				newObjectField.getUserId(), newObjectField.getObjectFieldId(),
-				objectStateFlow.getObjectStates());
+			if (oldObjectField != null) {
+				ObjectStateFlow serviceBuilderObjectStateFlow =
+					_objectStateFlowLocalService.
+						fetchObjectFieldObjectStateFlow(
+							oldObjectField.getObjectFieldId());
+
+				_objectStateFlowLocalService.updateObjectStateFlowTransitions(
```

Comment:
I believe that would be better if we break this method in two. One for delete and one for add the the transitions. What do you think?

Reply:
We saw [this examples](https://github.com/liferay-objects/liferay-portal/blob/fd512b62535d52ff29cc169abc988ab09f2cf20f/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectStateFlowLocalServiceImpl.java#L157-L184) and decided to follow with the current approach.